### ISSUE TITLE
Insteon (setup page on HA for Insteon)

### ISSUE THREAD

**Issue by TangoOversway** [home-assistant/home-assistant.github.io#19537]
### Feedback

For this page:  https://www.home-assistant.io/integrations/insteon/

The page says that Insteon 2448A7 USB Adaptor works with HA. I bought one. I've plugged it in. Then I checked HA - it has not discovered the adaptor. I have searched and can't find anything about how to start AutoDiscovery. I started trying to add Insteon as a new integration on HA, but I'm not sure what to add. I know it's not a PLM - I have one of those for an old system. I guessed it's the "Hub Version 2." Okay, but now it needs the IP address, the port number and username and password.

From this page, I have no idea how to figure out what info to provide HA for this. I have no idea how to run AutoDiscovery (again, searched for it).

One reason I went with HA is because it did so many different types of systems, like Z-Wave, Zigbee, Insteon, and others. But at this point, even after using Google, I find forum pages that link back to this page for how to set up Insteon.

I'm a retired programmer, I'm not a technophobe, but I have no idea how to add my 2448A7 USB Adaptor to HA, even though it's supposed to work with HA.

### URL

https://www.home-assistant.io/integrations/insteon/

### Version

2021.9.7

**Comment by teharris1** [comment]
Auto discovery is not available for Insteon. You need to go to `Configuration -> Integrations` and select `Instoen`. This wizard will walk you through the setup. 2448A7 USB Adaptor is treated as a PLM. The key item you need to know is the USB port that the modem is connected to. Hope that helps.

**Comment by TangoOversway** [comment]
Okay, that’s helpful! Thank you for such a quick reply!

Any tips on how to find the path to the USB port? And won’t that change if I change my USB devices? Right now I have 3 USB devices on my Raspberry Pi and I’m thinking I might get a USB hub so they’re spaced out more and in case I want to add Zigbee or more protocols I don’t have. I know when I was setting up RetroPie on a Pi 3, I found if I removed plugs, I had to always make sure they were put back in the same USB connectors or the devices would get mixed up and some functions would not work.

When I’m done, I do plan to post what I did, step by step, on the HA forum, so it’ll be out there and easy to follow for anyone else.


Hal

**Comment by teharris1** [comment]
If you are using HASS.IO on your Rpi then you have access to the USB info using `Supervisor` and selecting the `System` tab at the top. Under the `Host` panel there is a menu in the lower right that will show you hardware info including USB paths.

![image](https://user-images.githubusercontent.com/9055260/135666168-8334199a-7db0-44aa-87bb-b26989f0df93.png)

Scroll down to `ttyUSBx` which will show you the USB devices connected to the Rpi.
![image](https://user-images.githubusercontent.com/9055260/135666472-839ad133-6113-464d-935d-b927a2cb32ac.png)


FYI, my 2448A7 is the one marked `ttuUSB2`.

**Comment by TangoOversway** [comment]
Thank you - I have it listed as installed.

I’m trying to add a device. Am I correct that there is no actual discovery for adding devices? It looks like I have to add a device and pick override and use the device address and category info.

I’ve added a dimmer this way, but I can’t be sure it’s really been added, since the controls in HA don’t change the dimmer. I click what I think is the off/on switch and it goes from off to on, but then back to off. Whether the lights are on or not, it doesn’t change them.

Also, when I added the USB PLM, I used the long address (in your case, that would be /dev/serial/by-id/usb-FTDI-FT232R_USB_UART_A900EWXY-if00-port0. That seemed to work fine and I would think if I just used /dev/ttyUSB2 instead, that the USB device node could change on reboot, but the serial/by-id node is going to stay the same as long as it’s there. (Unless I add another!)

I think I said this - I’m also making notes so I can write this up and post it in the HA forum as a guide for anyone in the future. Do you mind if I use your screenshots? I will, of course, give you credit for them and for the help!



Hal

**Comment by teharris1** [comment]
Adding a device is something I have been meaning to make a lot easier but here are the steps:
1. Go to `Developer Tools` and click on the `Services` tab at the top. 
2. In the `Service` box select `Insteon: Add all link`
3. Fill in the form with the following:
  `Group: 0`
  `Mode: controller`
4. Press the `Call Service` button

This will place your modem in linking mode. You then have about 3 minutes to press the `Set` button on your device. This will link the device to the modem. Once the two are linked, the Insteon module will display the device in Home Assistant.

**Comment by TangoOversway** [comment]
I hate to keep bothering you on this, but I’m not getting much help on the HA board. I think this is outside of most peoples’ experience.

I don’t want to drag things on and on, but I don’t know enough about how the system works to know what details are important and what you can skip over, so I hope it’s easy to pull out the important points on this and skip the rest!

I have found someone quite helpful on the Insteon forum. Before I saw this, today (it didn’t show up before - I don’t know why!), I was going to Insteon as a service and using the Override to add devices. When I did that, HA, on boot, would see the device status and use that, but I couldn’t control the device and any changes I made (like turning lights on and off) did not show up in HA. HA would only get the state on boot - after that, nothing.

I was concerned about RF issues, since the 2448A7 is USB and I don’t know how much power it’s drawing. I have a 2413S AC powered PLM that I was using on my old ISY 994i system. (I really don’t want to use the ISYs anymore for a number of reasons - I’m not saying they’re bad, I just don’t want to use them at this point.) I did a factor reset on the 2413S and plugged it in so the 2448A7, the 2413S, and the Micro Dimmer I’m using for testing were all in the same room. (People have told me the 2413S would relay whatever signals it detected AND, since it’s dual band, it would send them through RF and the power lines.

Having it in the same room, plugged in, was no help. So I got a DB9S connector, used a USB to serial adaptor, and connected the 2413S to my Pi with HAOS on it. I deleted Insteon, rebooted, and then set it up using the 2413S as my PLM. (The 2448A7 was out.) Again, and you probably know this, the override add did not work. It makes it look like the device is there, but there’s no communication, except when I boot HA.

That’s when I thought to check and see if you had replied and I had not seen it.

I did this, with both the 2413s and the 2448A7 (I really wish Insteon would add human names to these things!):
- I deleted Insteon as an integration
- I rebooted HAOS
- I added Insteon as an integration.
- I followed your instructions as quoted below (calling the service to add all nodes/devices)

It’s worth noting that I’ve added a powered USB hub to the Pi 4 running HAOS and when I put the 2448A7 in to that, and entered the USB address for it (I use the by-id path, since that won’t change on rebooting), HA could not find it. HA could only find the 2448A7 when it was plugged directly into the Pi.

Each time I went through this process and hit "Call service," the blue button changed to green, then back to blue. It didn’t respond in any other way (like informing me it was trying to link or did link). After doing this, and going back to Integrations, I did not see another device added to Insteon.

Also one important question: Can I actually add a device by using the forced override? There are times that’d be a LOT easier. For instance, my garage door opener device from Insteon is rather high up and it’ll be a while before I can lug an A type ladder up to the house to press the set button and sometimes the Micro Dimmers are inside wall boxes.

Thank you for your time - I really appreciate it and I will be sure I have things written up clearly on a post on the forum to help others.



Hal

**Comment by TangoOversway** [comment]
SUCCESS!

I tried a number of combinations and found that the 2448A7 wasn’t able to do it on its own. I don’t have 100% solid proof, but my opinion is that it has a weak signal. I finally shut HA down, plugged in the 2413S, and rebooted and things were working.

So, at least in my setup, the 2448A7 was not working on its own, it needs the 2413S to relay signals. All three were in the same room, a 2 floor great room. There is a stone fireplace chimney between the 2448A7 and the Micro Dimmer, but I would think there would still be enough RF "bounce" in the room that it wouldn’t be an issue. It’s not a problem when I tested with my phone and a bluetooth headset. I have considered redoing the setup and connecting the 2413S with my cable, but, for now, I’d rather not. Maybe when I move the setup into my "tech closet," I will. It’s just easier when the PLM doesn’t have to be connected directly to the Pi. (On the other hand, if I test it and the 2413S works fine without the 2448A7, I may return it and just keep the AC powered PLM.

I’ll write this info up and post it so it’s clearer for people to set up Insteon. Is it okay if I credit you with the information?


Thank you!



Hal
